framework/console/controllers/MigrateController.php

```diff
@@ -268,28 +264,36 @@ protected function generateMigrationSourceCode($params)
             $templateFile = $this->generatorTemplateFiles['drop_column'];
             $table = mb_strtolower($matches[2], Yii::$app->charset);
         } elseif (preg_match('/^create_(.+)$/', $name, $matches)) {
-            $this->addDefaultPrimaryKey();
+            $this->addDefaultPrimaryKey($fields);
             $templateFile = $this->generatorTemplateFiles['create_table'];
             $table = mb_strtolower($matches[1], Yii::$app->charset);
         } elseif (preg_match('/^drop_(.+)$/', $name, $matches)) {
-            $this->addDefaultPrimaryKey();
+            $this->addDefaultPrimaryKey($fields);
             $templateFile = $this->generatorTemplateFiles['drop_table'];
             $table = mb_strtolower($matches[1], Yii::$app->charset);
         }
 
         return $this->renderFile(Yii::getAlias($templateFile), array_merge($params, [
             'table' => $table,
-            'fields' => $this->fields,
-            'foreignKeys' => $this->foreignKeys,
+            'fields' => $fields,
+            'foreignKeys' => $foreignKeys,
         ]));
     }
 
     /**
      * Parse the command line migration fields
+     * @return array parse result with following fields:
+     *
+     * - fields: array, parsed fields
+     * - foreignKeys: array, detected foreign keys
+     *
      * @since 2.0.7
      */
     protected function parseFields()
```

Comment:
Related to #11287  since this method no longer needs to get called during `beforeAction` can we make it take a parameter `$table` to behave similar to https://github.com/yiisoft/yii2/pull/11287/files#diff-952bc7045bade5b9e9c44621c5f93c19R335 ?